lms/static/scripts/frontend_apps/components/dashboard/OrderableActivityTable.tsx

```diff
@@ -0,0 +1,79 @@
+import type { DataTableProps, Order } from '@hypothesis/frontend-shared';
+import { DataTable } from '@hypothesis/frontend-shared';
+import { useOrderedRows } from '@hypothesis/frontend-shared';
+import type { OrderDirection } from '@hypothesis/frontend-shared/lib/types';
+import { useMemo, useState } from 'preact/hooks';
+
+import type { BaseDashboardStats } from '../../api-types';
+
+export type OrderableActivityTableProps<T extends BaseDashboardStats> = Pick<
+  DataTableProps<T>,
+  'emptyMessage' | 'rows' | 'renderItem' | 'loading' | 'title'
+> & {
+  columnNames: Partial<Record<keyof T, string>>;
+  defaultOrderField: keyof T;
+};
+
+/**
+ * List of columns which should start sorted in descending order
+ */
+const descendingOrderColumns: ReadonlyArray<string> = [
```

Comment:
`readonly string[]` can also be used. I would guess it is slightly more idiomatic to use the custom syntax, but I haven't looked into it.